sixtyfps_compiler/widgets/ugly/sixtyfps_widgets.60

```diff
@@ -308,6 +308,68 @@ export GroupBox := GridLayout {
     }
 }
 
+export TabWidgetImpl := Rectangle {
+    property <length> content_x: 0;
+    property <length> content_y: tabbar_preferred_height;
+    property <length> content_height: height - tabbar_preferred_height;
+    property <length> content_width: width;
+    property <length> tabbar_x: 0;
+    property <length> tabbar_y: 0;
+    property <length> tabbar_height: tabbar_preferred_height;
+    property <length> tabbar_width: width;
+
+    property <length> tabbar_preferred_height;
+    property <length> tabbar_preferred_width;
+    property <length> content_min_height;
+    property <length> content_min_width;
+    property <int> current_index;
+
+    border_width: 1px;
+    border_color: Palette.border_color;
+    border_radius: 2px;
+
+    preferred_width: content_min_width;
+    min_width: content_min_width;
+    preferred_height: content_min_height + tabbar_preferred_height;
+    min_height: content_min_height + tabbar_preferred_height;
+
+}
+export TabImpl := Rectangle {
+    property<string> title <=> t.text;
+    //property<image> icon;
+    property<bool> enabled : true;
+    property<bool> pressed;
+    property<int> current;
+    property<int> tab_index;
+    property<int> num_tabs;
```

Comment:
This seems unused in both implementations. Does the tab itself need that information?

Reply:
the Qt implementation needs it because it might draw the last tab differently